README.md

```diff
@@ -43,27 +43,29 @@ import (
     "github.com/Eppo-exp/golang-sdk/v4/eppoclient"
 )
 
-var eppoClient = &eppoclient.EppoClient{}
+var eppoClient *eppoclient.EppoClient
 
 func main() {
     assignmentLogger := NewExampleAssignmentLogger()
 
-    eppoClient = eppoclient.InitClient(eppoclient.Config{
+    eppoClient, err = eppoclient.InitClient(eppoclient.Config{
         SdkKey:           "<your_sdk_key>",
         AssignmentLogger: assignmentLogger,
     })
+    if err != nil {
+        log.Fatalf("Failed to initialize Eppo client: %v", err)
+    }
```

Comment:
Allow the caller to handle failures to initialize as they desire - it is obvious from the compiler type that an `error` is returned.

👉  We could consider this a breaking API change and publish a v5.

Reply:
Yeah with this signature being different it's probably v5

Reply:
+1 to publishing a new major version according to semver